packages/pigeon/lib/linux_generator.dart

```diff
@@ -0,0 +1,1325 @@
+          '}', () {
+        _writeObjectNew(indent, module, responseName);
+        indent.writeln('self->value = fl_value_new_list();');
+        indent.writeln(
+            'fl_value_append_take(self->value, fl_value_new_string(code));');
+        indent.writeln(
+            'fl_value_append_take(self->value, fl_value_new_string(message));');
+        indent.writeln('fl_value_append(self->value, details);');
+        indent.writeln('return self;');
+      });
+    }
+
+    indent.newln();
+    _writeObjectStruct(indent, module, api.name, () {
+      indent.writeln('FlBinaryMessenger* messenger;');
+      indent.writeln('const ${className}VTable* vtable;');
+      indent.writeln('gpointer user_data;');
+      indent.writeln('GDestroyNotify user_data_free_func;');
+
+      indent.newln();
+      for (final Method method in api.methods) {
+        final String methodName = _snakeCaseFromCamelCase(method.name);
+        indent.writeln('FlBasicMessageChannel* ${methodName}_channel;');
+      }
+    });
+
+    indent.newln();
+    _writeDefineType(indent, module, api.name);
+
+    for (final Method method in api.methods) {
+      final String methodName = _snakeCaseFromCamelCase(method.name);
+      final String responseName = _getResponseName(api.name, method.name);
+      final String responseClassName = _getClassName(module, responseName);
+
+      indent.newln();
+      indent.writeScoped(
+          'static void ${methodName}_cb(FlBasicMessageChannel* channel, FlValue* message, FlBasicMessageChannelResponseHandle* response_handle, gpointer user_data) {',
+          '}', () {
+        _writeCastSelf(indent, module, api.name, 'user_data');
+
+        indent.newln();
+        indent.writeScoped(
+            'if (self->vtable == nullptr || self->vtable->$methodName == nullptr) {',
+            '}', () {
+          indent.writeln('return;');
+        });
+
+        final List<String> methodArgs = <String>[
+          for (int i = 0; i < method.parameters.length; i++)
+            _fromFlValue(module, method.parameters[i].type,
+                'fl_value_get_list_value(message, $i)'),
+        ];
+
+        indent.newln();
+        if (method.isAsynchronous) {
+          final List<String> vfuncArgs = <String>['self'];
+          vfuncArgs.addAll(methodArgs);
+          vfuncArgs.addAll(<String>['response_handle', 'self->user_data']);
+          indent.writeln("self->vtable->$methodName(${vfuncArgs.join(', ')});");
+        } else {
+          final List<String> vfuncArgs = <String>['self'];
+          vfuncArgs.addAll(methodArgs);
+          vfuncArgs.add('self->user_data');
+          indent.writeln(
+              "g_autoptr($responseClassName) response = self->vtable->$methodName(${vfuncArgs.join(', ')});");
+          indent.writeScoped('if (response == nullptr) {', '}', () {
+            indent.writeln(
+                'g_warning("No response returned to ${api.name}.${method.name}");');
```

Comment:
We should compose this warning and the warning below dynamically, so that we have:
- one copy of the base of this message,
- one copy of the base of the other message, and
- one copy of each API+method name.

Right now this is creating a bunch of long strings per method, adding binary size, for a case that pretty much shouldn't happen.